sdk/identity/Azure.Identity/src/CredentialDiagnosticScope.cs

```diff
@@ -34,27 +36,26 @@ public AccessToken Succeeded(AccessToken token)
             return token;
         }
 
-        public AuthenticationFailedException FailAndWrap(Exception ex)
+        public Exception WrapAndThrow(Exception ex)
```

Comment:
Perhaps I'm being a bit pedantic here but I think the name should be `FailAndThrow` rather than `WrapAndThrow`. My reasoning behind this is that this is an instance method on `CredentialDiagnosticScope` and the high order bit is that we are "failing" the diagnostic scope and then we are throwing the exception (possibly wrapped). If you want to be most precise you could also call it `FailWrapAndThrow` but this might be a bit long winded.

Reply:
Fixed